packages/page-settings/src/modals/NetworkSpecs.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017-2020 @polkadot/app-settings authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { getSystemChainColor } from '@polkadot/apps-config/ui/general';
+import { Button, Input, Modal } from '@polkadot/react-components';
+import { BareProps } from '@polkadot/react-components/types';
+import { useApi, useDebounce } from '@polkadot/react-hooks';
+import registry from '@polkadot/react-api/typeRegistry';
+import ApiPromise from '@polkadot/api/promise';
+import { createType } from '@polkadot/types';
+import addressDefaults from '@polkadot/util-crypto/address/defaults';
+import { QrNetworkSpecs } from '@polkadot/react-qr';
+import { NetworkSpecsStruct } from '@polkadot/ui-settings';
+
+import React, { useEffect, useReducer, useState } from 'react';
+import styled from 'styled-components';
+
+import { useTranslation } from '../translate';
+
+interface Props extends BareProps {
+  onClose: () => void;
+}
+
+function getRandomColor (): string {
+  const letters = '0123456789ABCDEF';
+  let color = '#';
+  for (let i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+}
+
+const buildNetworkSpecs = async (api: ApiPromise, systemChain: string, systemName: string): Promise<Partial<NetworkSpecsStruct>> => {
+  const [properties, blockHash] = await Promise.all([
+    api.rpc.system.properties(),
```

Comment:
This is already available on the registry, `registry.ss58Format`, etc.

Reply:
Exposed the stuff I use for the extension as a hook - you can just pull the fields from there, then we don't have duplication.

https://github.com/polkadot-js/apps/pull/2460

Reply:
We just need to add color in there, that is missing for you.